torch/csrc/jit/runtime/interpreter/code_impl.h

```diff
@@ -59,15 +59,26 @@ struct WithCurrentNode {
   Node* old_value_;
 };
 
+
+struct NodeSourceInfo {
+  const char* func_name_;
```

Comment:
You should use use nullptr instead of NULL + optional<std::string> for less chance accidental bugs / mem leaks.